pkg/registry/core/node/strategy.go

```diff
@@ -131,22 +139,30 @@ func (nodeStatusStrategy) PrepareForCreate(ctx context.Context, obj runtime.Obje
 	node := obj.(*api.Node)
 	// Nodes allow *all* fields, including status, to be set on create.
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.DynamicKubeletConfig) {
-		node.Status.Config = nil
-	}
+	node.Status.Config = nil
 }
 
 func (nodeStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
 	newNode := obj.(*api.Node)
 	oldNode := old.(*api.Node)
 	newNode.Spec = oldNode.Spec
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.DynamicKubeletConfig) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DynamicKubeletConfig) && !nodeStatusConfigInUse(oldNode) {
 		newNode.Status.Config = nil
-		oldNode.Status.Config = nil
 	}
 }
 
+// nodeConfigSourceInUse returns true if node's Spec ConfigSource is set(used)
```

Comment:
Update comment and logic to reflect status

Reply:
done